### ISSUE TITLE
Add test that checks FDD can be published without nuget.org artifacts.

### ISSUE THREAD

**Issue by tmds** [redhat-developer/dotnet-regular-tests#142]
We should add a test that checks a FDD application can be published without contacting nuget.org.

Surprisingly, https://github.com/redhat-developer/s2i-dotnetcore/issues/358 indicates this was not the case with .NET Core 3.1.0.

The test is expected to pass with the latest versions of 3.1 and 5.0 sdk.

cc @omajid @RheaAyase

**Comment by RheaAyase** [comment]
Thoughts @HoundThe? Do you want to take on some of these issues?

**Comment by HoundThe** [comment]
I can try, but I am not sure If I understand the task :D

**Comment by omajid** [comment]
Here's my understanding.

Goal: We want a test that verifies that [users can publish their applications](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) in the [framework-dependent mode](https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-framework-dependent) and not have to be connected to the internet/nuget.org.

Implementation Ideas: some possible options, not exhaustive

- We can set up an environment that can't access nuget.org (or the entire internet) and then run the publish-in-framework-dependent-mode command. The command should work offline and produce a working application.

- We can run publish-in-framework-dependent-mode command under some sort of tracing tool to verify that we dont need/connect to nuget.org

**Comment by tmds** [comment]
> Implementation Ideas: some possible options

Another option:

The default config is to use `nuget.org` as a source, so it goes unnoticed when packages are consumed.

To get rid of the source, you can add a `NuGet.Config` file and `</clear>` the sources.
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
  </packageSources>
</configuration>
```

That is how the container-tool gets built in https://github.com/redhat-developer/s2i-dotnetcore/blob/master/5.0/build/s2i/bin/container-tool.

I assume Microsoft has good testing that ensures the `<clear/>` thing works as expected.

**Comment by HoundThe** [comment]
Does something like this make sense then <https://github.com/HoundThe/dotnet-regular-tests/commit/b14750237dff699c43edd62e62c736de891a7f48> ? I am not sure as I can't reproduce the error even with disabled internet, I'll try to get the 3.1.0 package to try it on.

**Comment by tmds** [comment]
> Does something like this make sense then HoundThe@b147502 ? 

@HoundThe yes, that is going in the right direction. You can make a PR, and we can give feedback using the usual github workflow. You can always do this, the PR can still go any direction based on the feedback.

> I am not sure as I can't reproduce the error even with disabled internet, I'll try to get the 3.1.0 package to try it on.

To test your test, try adding a package to the project that surely must come from nuget.org. Then your test should fail.

**Comment by HoundThe** [comment]
Oh yea it works, I've cleared the NuGet cache and with the NuGet package, it fails to restore when the <clear/> option is set., thanks! I'll make the PR.
